core/arch/arm/kernel/vfp_a32.S

```diff
@@ -4,20 +4,30 @@
  */
 
 #include <asm.S>
+#include <arm32.h>
+#include <generated/arm32_sysreg.S>
 
 	.fpu	neon
 
 /* void vfp_save_extension_regs(uint64_t regs[VFP_NUM_REGS]); */
 FUNC vfp_save_extension_regs , :
-	vstm	r0!, {d0-d15}
-	vstm	r0, {d16-d31}
+	mov	r1, r0
```

Comment:
We could save one instruction by using `r0` directly.
`read_cpacr` can operate with `r1` instead.

Reply:
`read_cpacr()` is just an assembly macro, the `reg` argument is the chosen destination register.
```
        .macro read_cpacr reg
        mrc p15, 0, \reg, c1, c0, 2
        .endm
```

Reply:
I check the wrong file (arm32_sysreg.h) which is inline assembly.
```
/* Coprocessor Access Control Register */
static inline __noprof uint32_t read_cpacr(void)
{
        uint32_t v;

        asm volatile ("mrc p15, 0, %0, c1, c0, 2" : "=r"  (v));

        return v;
}
```

I'll simplify the patch as you suggested. Thanks !